test/semantics/modfile24.f90

```diff
@@ -63,7 +63,7 @@ subroutine s(a, b, n)
 !Expect: m3.mod
 !module m3
 ! type::t
-!  real(4)::c[1_8:10_8,1_8:*]
```

Comment:
I think component `c` might be violating C746. As I happen to be working on coarray implications in ALLOCATE, I fall upon it today. It applies only for derived type components:

>  If a coarray-spec appears, it shall be a deferred-coshape-spec-list and the component shall have the ALLOCATABLE attribute.

So you might get an error here when we check more coarray constraints.
As I understand it you should have: `real, allocatable :: c(:)[:,:]`

Reply:
Thanks for catching that -- I'll fix it.